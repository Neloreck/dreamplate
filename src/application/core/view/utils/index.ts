export * from "@/core/view/utils/rendering";
