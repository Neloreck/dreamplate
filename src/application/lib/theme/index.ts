export * from "@/lib/theme/types";
export * from "@/lib/theme/utils";
