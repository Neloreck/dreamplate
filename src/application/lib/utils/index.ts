export * from "@/lib/utils/local_storage";
export * from "@/lib/utils/delay";
