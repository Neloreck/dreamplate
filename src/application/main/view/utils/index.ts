import { LazyLoadComponentFactory } from "@Main/view/utils/lazy_load/LazyLoadComponentFactory";

export const lazyLoadComponentFactory: LazyLoadComponentFactory = new LazyLoadComponentFactory();

export * from "@Main/view/utils/animations";
export { LazyLoadComponentFactory } from "@Main/view/utils/lazy_load/LazyLoadComponentFactory";
