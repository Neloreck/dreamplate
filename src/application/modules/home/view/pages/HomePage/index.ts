export { HomePage, IHomePageInjectedProps } from "./HomePage.Component";
