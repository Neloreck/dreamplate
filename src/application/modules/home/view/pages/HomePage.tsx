import { useManager } from "dreamstate";
import { ReactElement, ReactNode } from "react";
import { createUseStyles } from "react-jss";

import { down } from "#/macroses/style.macro";
import { RouterManager } from "@/core/data/store";
import { MainHeader } from "@/core/view/components";
import { IApplicationTheme } from "@/lib/theme";

export const useStyles = createUseStyles(({ palette, spacing }: IApplicationTheme) => ({
  link: {
    "& a": {
      color: palette.text.primary
    },
    backgroundColor: palette.background.paper,
    display: "flex",
    justifyContent: "center",
    margin: spacing.unit * 2,
    padding: spacing.unit * 2,
    minWidth: spacing.unit * 15
  },
  content: {
    alignItems: "center",
    alignSelf: "stretch",
    display: "flex",
    flexDirection: "column",
    flexGrow: 1,
    justifyContent: "center",
    padding: spacing.unit * 2,
    overflowY: "auto",
    overflowX: "hidden"
  },
  linksList: {
    [down("sm")]: {
      flexDirection: "column"
    },
    display: "flex",
    flexDirection: "row",
    padding: spacing.unit * 10
  }
}));

const renderReferenceCard = (className: string, label: string, href: string): ReactNode => (
  <div className={className}>
    <a href={href} target={"_blank"} rel={"noopener noreferrer"}>
      {label}
    </a>
  </div>
);

export function HomePage({
  classes: { content, link, linksList } = useStyles(),
  routerContext: { history } = useManager(RouterManager)
}): ReactElement {
  return (
    <>
      <MainHeader />

      <main className={content}>
        <div className={linksList}>
          {renderReferenceCard(link, "React", "https://reactjs.org/")}
          {renderReferenceCard(link, "Typescript", "https://www.typescriptlang.org/")}
          {renderReferenceCard(link, "DreamState", "https://github.com/Neloreck/dreamstate/")}
        </div>

        <a href={history.createHref("/about")}>
          <button>About</button>
        </a>
      </main>
    </>
  );
}
