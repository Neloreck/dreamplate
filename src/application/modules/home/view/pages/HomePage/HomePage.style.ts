/**
 * @module @application/home
 */

import { createUseStyles } from "react-jss";

// Lib.
import { down, IApplicationTheme } from "@Lib/theme";

export const useStyles = createUseStyles(({ palette, spacing }: IApplicationTheme) => ({
  cardLink: {
    "& a": {
      color: palette.text.primary,
    },
    backgroundColor: palette.background.paper,
    display: "flex",
    justifyContent: "center",
    minWidth: spacing.unit * 15,
    transitionDuration: "500ms"
  },
  content: {
    alignItems: "center",
    alignSelf: "normal",
    backgroundColor: palette.background.default,
    backgroundSize: "cover",
    display: "flex",
    flexDirection: "column",
    flexGrow: 1,
    justifyContent: "center",
    overflowY: "auto",
    transitionDuration: "350ms"
  },
  linkCard: {
    [down("sm")]: {
      flexDirection: "column"
    },
    display: "flex",
    flexDirection: "row",
    padding: spacing.unit * 10
  }
}));
