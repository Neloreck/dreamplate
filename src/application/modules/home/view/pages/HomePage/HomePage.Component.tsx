import * as React from "react";
import { PureComponent, ReactNode } from "react";

// Lib.
import { Styled } from "@Lib/decorators";

// View.
import { IMainHeaderInjectedProps, MainHeader } from "@Main/view/components/MainHeader";
import { Card, Grid, WithStyles } from "@material-ui/core";
import { homePageStyle } from "./HomePage.Style";

// Props.
export interface IHomePageOwnProps {}
export interface IHomePageInjectedProps extends WithStyles<typeof homePageStyle> {}
export interface IHomePageProps extends IHomePageOwnProps, IHomePageInjectedProps {}

@Styled(homePageStyle)
export class HomePage extends PureComponent<IHomePageProps> {

  public render(): ReactNode {

    const { classes } = this.props;

    return (
      <Grid
        className={classes.root}
        direction={"column"}
        wrap={"nowrap"}
        container
      >

       <MainHeader {...{} as IMainHeaderInjectedProps}/>

        <Grid
          className={classes.content}
          justify={"center"}
          direction={"column"}
          alignItems={"center"}
          component={"main"}
          container
        >

          <Card className={classes.card}>
            Change some code and move on.
          </Card>

          <Grid direction={"row"} justify={"center"} container>

            <Card className={classes.card}>
              <a target={"_blank"} href={"https://reactjs.org/"}>
                React
              </a>
            </Card>

            <Card className={classes.card}>
              <a target={"_blank"} href={"https://material-ui.com/"}>
                MaterialUI
              </a>
            </Card>

            <Card className={classes.card}>
              <a target={"_blank"} href={"https://www.typescriptlang.org/"}>
                Typescript
              </a>
            </Card>

            <Card className={classes.card}>
              <a target={"_blank"} href={"https://github.com/Neloreck/dreamstate/"}>
                DreamState
              </a>
            </Card>

          </Grid>

        </Grid>

      </Grid>
    );
  }

}
