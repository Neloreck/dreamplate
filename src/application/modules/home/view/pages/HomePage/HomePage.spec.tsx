import "@Test/utils/mock_custom_components";

// Test.
import { getThemedComponent, withMockedTheme } from "@Lib/theme/testing";
import { shallow } from "enzyme";

// View.
import { HomePage, IHomePageInjectedProps } from "./HomePage.component";

describe("Home page component rendering.", () => {

  it("Should match HOC snapshot.", () => {

    const wrapper = shallow(<HomePage {...{} as IHomePageInjectedProps}/>);

    expect(wrapper).toMatchSnapshot("Default HOC layout.");
  });

  it("Should match mounted snapshot with default theme.", () => {

    const wrapper = shallow(withMockedTheme(<HomePage {...{} as IHomePageInjectedProps}/>));
    const mounted = getThemedComponent(wrapper);

    expect(mounted).toMatchSnapshot("Mounted with default theme.");
  });
});
