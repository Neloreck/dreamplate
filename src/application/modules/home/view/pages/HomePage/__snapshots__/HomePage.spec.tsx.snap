// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page component rendering. Should match HOC snapshot.: Default HOC layout. 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Home page component rendering. Should match mounted snapshot with default theme.: Mounted with default theme. 1`] = `
<HomePage
  classes={
    Object {
      "card": "HomePage-card-0-2-1",
      "content": "HomePage-content-0-2-2",
    }
  }
>
  <JssContextSubscriber>
    <WithStyles(MainHeader)
      innerRef={null}
      jssContext={
        Object {
          "classNamePrefix": "",
          "disableStylesGeneration": false,
        }
      }
      theme={
        Object {
          "palette": Object {
            "background": Object {
              "default": "#FFF",
              "paper": "#eeeeee",
            },
            "primary": Object {
              "dark": "#7986cb",
              "light": "#d1c4e9",
              "main": "#9fa8da",
            },
            "secondary": Object {
              "dark": "#ab47bc",
              "light": "#7986cb",
              "main": "#ba68c8",
            },
            "text": Object {
              "primary": "#000",
              "secondary": "#444444",
            },
            "type": "light",
          },
          "spacing": Object {
            "unit": 4,
          },
        }
      }
    >
      <MainHeader
        classes={
          Object {
            "root": "MainHeader-root-0-2-3",
          }
        }
      >
        <header
          className="MainHeader-root-0-2-3"
        >
          <custom-heading
            size={3}
            text="Boilerplate"
          />
          <custom-button
            label="Toggle"
            onClick={[Function]}
          />
        </header>
      </MainHeader>
    </WithStyles(MainHeader)>
  </JssContextSubscriber>
  <main
    className="HomePage-content-0-2-2"
  >
    <content-card
      class="HomePage-card-0-2-1"
    >
      <a
        href="https://reactjs.org/"
        rel="noreferrer"
        target="_blank"
      >
        React
      </a>
    </content-card>
    <content-card
      class="HomePage-card-0-2-1"
    >
      <a
        href="https://www.typescriptlang.org/"
        rel="noreferrer"
        target="_blank"
      >
        Typescript
      </a>
    </content-card>
    <content-card
      class="HomePage-card-0-2-1"
    >
      <a
        href="https://github.com/Neloreck/dreamstate/"
        rel="noreferrer"
        target="_blank"
      >
        DreamState
      </a>
    </content-card>
  </main>
</HomePage>
`;
